website_url_sitemap_exclusion/models/website.py

```diff
@@ -0,0 +1,29 @@
+#  Copyright 2023 Simone Rubino - TAKOBI
+#  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import re
+from odoo import api, fields, models
+
+
+class Website(models.Model):
+    _inherit = 'website'
+
+    forbidden_url_ids = fields.One2many(
+        comodel_name="website.forbidden.url",
+        inverse_name="website_id",
+        string="Forbidden URLs regex",
+    )
+
+    @api.multi
+    def enumerate_pages(self, query_string=None, force=False):
+        pages = super().enumerate_pages(query_string=query_string, force=force)
+        forbidden_url_regexes = self.mapped("forbidden_url_ids.regex")
+        forbidden_url_compiled_regexes = list(map(re.compile, forbidden_url_regexes))
```

Comment:
```suggestion
        forbidden_url_compiled_regexes = map(re.compile, forbidden_url_regexes)
```
no need to convert to list

Reply:
It is actually needed to convert to a `list` because otherwise the `map` generator is exhausted after the first loop.
See how the variable `m` is empty after the first loop:
```python
>>> l = [1, 2, 3]
>>> m = map(print, l)
>>> for p in m:
...     pass
... 
1
2
3
>>> for p in m:
...     pass
... 
>>>
```